### DISCUSSION TITLE
请问下siamrpn里面也有训练代码吗？我只看到了rpn++的

### DISCUSSION THREAD

**Discussion by Dennisky** [STVIR/pysot#411] [General] [answered]

**Comment by ZhiyuanChen** [comment, marked as answer]
用AlexNet backbone，别用depthwise head，你就得到了SiamRPN。

**Comment by wWHWw** [comment]
by changing RPN.TYPE in siamrp_alex_dwxcorr/config.yaml from ’DepthwiseRPN‘ to 'UPChannelRPN' ？ @zyc-ai

**Comment by ZhiyuanChen** [comment]
> by changing RPN.TYPE in siamrp_alex_dwxcorr/config.yaml from ’DepthwiseRPN‘ to 'UPChannelRPN' ？ @zyc-ai


Precisely

And don't forget the sample strategy

**Comment by wWHWw** [comment]
> > by changing RPN.TYPE in siamrp_alex_dwxcorr/config.yaml from ’DepthwiseRPN‘ to 'UPChannelRPN' ？ @zyc-ai
> 
> Precisely
> 
> And don't forget the sample strategy

you mean do not use sample strategy from DaSianRPN?
Set  __C.DATASET.NEG to 0 ,so when building dataset

```
        if neg:
            template = dataset.get_random_target(index)
            search = np.random.choice(self.all_dataset).get_random_target()
        else:
            template, search = dataset.get_positive_pair(index)
```
only get_positive_pair() will be used?
By the way , I also have some question about trainnig RPN with ytbb datasets.
I opened an issue  #415 here. Perhaps you can take a look at your convenience. 
Thanks a lot.@zyc-ai

**Comment by wWHWw** [comment]
And it seems like that in_channels=256 and out_channels=256 in the config.yaml will not be used in training SiamRPN?

**Comment by ZhiyuanChen** [comment]
> you mean do not use sample strategy from DaSianRPN?
> Set __C.DATASET.NEG to 0 ,so when building dataset
> 
> ```
>         if neg:
>             template = dataset.get_random_target(index)
>             search = np.random.choice(self.all_dataset).get_random_target()
>         else:
>             template, search = dataset.get_positive_pair(index)
> ```
> 
> only get_positive_pair() will be used?

Yes that's what I meant, but not by setting the negative sample to zero. The sample strategy of DaSiamRPN is to add object of different class as negative sample, it is important to have enough amount of negative smaple or it would be very unlikely for the network to work.

**Comment by ZhiyuanChen** [comment]
> And it seems like that in_channels=256 and out_channels=256 in the config.yaml will not be used in training SiamRPN?

Some of the input is hard coded which should be fixed.....

**Comment by wWHWw** [comment]
> > you mean do not use sample strategy from DaSianRPN?
> > Set __C.DATASET.NEG to 0 ,so when building dataset
> > ```
> >         if neg:
> >             template = dataset.get_random_target(index)
> >             search = np.random.choice(self.all_dataset).get_random_target()
> >         else:
> >             template, search = dataset.get_positive_pair(index)
> > ```
> > 
> > 
> > only get_positive_pair() will be used?
> 
> Yes that's what I meant, but not by setting the negative sample to zero. The sample strategy of DaSiamRPN is to add object of different class as negative sample, it is important to have enough amount of negative smaple or it would be very unlikely for the network to work.

I get that if i want to use distractor-aware strategy, negative sample shouldn't be set to zero.
But if I want train origin SiamesRPN model, is negative sample still needed ?

**Comment by ZhiyuanChen** [comment]
> But if I want train origin SiamesRPN model, is negative sample still needed ?

Yes, they are needed.

**Comment by wWHWw** [comment]
> > But if I want train origin SiamesRPN model, is negative sample still needed ?
> 
> Yes, they are needed.

So for the beginner like me , if I want train SiamRPN, i‘d better set this parameter as default.(__C.DATASET.NEG = 0.2)

**Comment by ZhiyuanChen** [comment]
> So for the beginner like me , if I want train SiamRPN, i‘d better set this parameter as default.(__C.DATASET.NEG = 0.2)

Yes.
